Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -139,3 +142,42 @@ void APSStarActor::SetStartTimeMenuStars()
 
 	StartTimeMenuStarsInternal = World->GetTimeSeconds();
 }
+
+//  Is get called when a Star actor is initialized
+void APSStarActor::OnInitialized(const FVector& PreviousActorLocation)
+{
+	const FPSRowData& CurrentRowData = UPSWorldSubsystem::Get().GetCurrentRow();
+
+	// set offset from previous if it's not first
+	// if the PreviousActorTransform is empty this means it is a first element and set init transform
+	if (PreviousActorLocation.Equals(FVector::ZeroVector))
+	{
+		SetActorTransform(CurrentRowData.StarActorTransform);
```

Comment:
You can simplify the code by removing the if-else block, since the same transformation is set in both branches. Additionally, it's more efficient to compute the desired transform once, instead of making separate SetActorLocation calls.

### Expected result
```cpp
FTransform DesiredTransform = CurrentRowData.StarActorTransform;

if (!PreviousActorLocation.Equals(FVector::ZeroVector))
{
    DesiredTransform.SetLocation(PreviousActorLocation + CurrentRowData.OffsetBetweenStarActors);
}

SetActorTransform(DesiredTransform);
```

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/bd5b21b7f15fa01f174c59cb316fe15a6a756755